testutil/wait.go

```diff
@@ -14,6 +14,24 @@ import (
 type testFn func() (bool, error)
 type errorFn func(error)
 
+func Wait(t *testing.T, test testFn) {
+	t.Helper()
+	retries := 500 * TestMultiplier()
+	for retries > 0 {
+		time.Sleep(10 * time.Millisecond)
+		retries--
+
+		success, err := test()
+		if success {
+			return
+		}
+
+		if retries == 0 {
+			t.Fatalf("timeout: %v", err)
+		}
+	}
+}
+
```

Comment:
I just got sick of writing the error callback. LMK if I should stop being lazy and switch to WaitForResult. :sweat_smile:

Reply:
I was tempted to write a similar helper recently, and I've had a few review comments on other folks' PRs of late where not returning an error in the `return false, nil` case ends up making for hard-to-debug test failures. There are ~600 uses of `WaitForResult*` helpers, so at some point I'd like to take a pass through and see how many we could move to this simpler helper... I think it's a lot.

Reply:
I'd also like to see this pattern cleaned up - but this particular helper doesn't prevent the `nil` error issue, and also I think the concept of `TestMultiplier` can be eliminated in favor of context/timeout

Reply:
What if I added 
```go
if err == nil {
	t.Fatalf("timeout waiting for test function to succeed (you should probably return a helpful error instead of nil!)")
}
```

The file name and line number will take you to the beginning of the test function that returned `false, nil`. Since the stack frame responsible for returning `false, nil` ... returned ... there's really nothing else we can do here.